src/main/java/gov/cabinetoffice/gapuserservice/service/SpotlightService.java

```diff
@@ -71,16 +72,20 @@ public String getAuthorizeUrl() throws Exception {
 
         String authorizationEndpoint = uri.toString();
 
-        state = generateRandomString(64);
+        String newState = generateRandomString(64);
```

Comment:
Not too familiar with these endpoints - but I'm struggling to understand how this would solve horizontally scaling. Wouldn't each instance continue to use this randomly generated string and still not match?

Reply:
we save the state in the db and then pull it back out when the user has competed the login flow with spotlight/salesforce, and check it's the same as the value we've been passed back. each new login will create a new state and invalidate any other logins in progress, but that was the case before (and I couldn't easily think of a way around it) - I think since this is something only superadmins can do, the likelihood of a two people logging in at once is vanishingly low.